client/types.go

```diff
@@ -197,12 +198,21 @@ type Session interface {
 	// Write value to the database for an ID
 	Write(namespace string, id string, t time.Time, value float64, unit xtime.Unit, annotation []byte) error
 
+	// WriteTagged value to the database for an ID and given tags.
+	WriteTagged(namespace string, id string, tags ident.TagIterator, t time.Time, value float64, unit xtime.Unit, annotation []byte) error
+
 	// Fetch values from the database for an ID
 	Fetch(namespace string, id string, startInclusive, endExclusive time.Time) (encoding.SeriesIterator, error)
 
 	// FetchAll values from the database for a set of IDs
 	FetchAll(namespace string, ids []string, startInclusive, endExclusive time.Time) (encoding.SeriesIterators, error)
 
+	// FetchTagged resolves the provided query to known IDs, and fetches the data for them.
+	FetchTagged(q Query, startInclusive, endExclusive time.Time) (encoding.SeriesIterators, error)
+
+	// FetchTaggedIDs resolves the provided query to known IDs.
+	FetchTaggedIDs(q Query, startInclusive, endExclusive time.Time) (encoding.SeriesIterators, error)
```

Comment:
Hm, we should change the return type yeah?  IDs will need to be `[]ts.ID` or something yeah?  Also do we want to include limits, etc?

Reply:
sure thing. 
- added a new iterator type in `client` for the first point,
- added corresponding fields for limit+pagetoken